### PR TITLE
Introduce hardening property to deny access to missing perms

### PR DESCRIPTION
**Pull request by tkountis**
Fixes https://github.com/hazelcast/hazelcast-enterprise/issues/1959
EE: https://github.com/hazelcast/hazelcast-enterprise/pull/1979

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java [hunk 1]
```diff
@@ -942,6 +942,15 @@
...
             new HazelcastProperty("hazelcast.executionservice.taskscheduler.remove.oncancel", false);
 
+    /**
+     * Client operations in a cluster are intercepted on the member side for permission control.
+     * Some API might not hold permission mappings, i.e., inherited JDK methods that are not yet supported, but still available
+     * to the end-user to run, in which case we used to allow execution. This property controls this behaviour, and by default
+     * (after 3.10) will reject any call to API that has no mapped permission.
+     */
+    public static final HazelcastProperty DENY_ACCESS_TO_MISSING_PERMISSION_MAPPING =
+            new HazelcastProperty("hazelcast.deny.access.to.missing.permission.mapping", false);
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/spi/properties/GroupProperty.java]: I think the doc, the prop name, and its default value are in contradiction. Either the default value should be `true` or the property name should be `hazelcast.allow.access.to.missing.permission.mapping`. I'd prefer the first.

**Comment by tkountis** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/config/SecurityConfig.java [hunk 2]
```diff
@@ -121,6 +125,15 @@ public SecurityConfig setMemberCredentialsConfig(CredentialsFactoryConfig creden
         return this;
     }
 
+    public boolean getClientBlockUnmappedActions() {
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/config/SecurityConfig.java]: Can you add some javadoc here and the same description in the `hazelcast-config-3.10.xsd` and `hazelcast-spring-3.10.xsd` XSD configs?

File: hazelcast/src/main/java/com/hazelcast/config/ConfigXmlGenerator.java [hunk 3]
```diff
@@ -248,6 +250,96 @@ private static void eventJournalXmlGenerator(XmlGenerator gen, Config config) {
...
+                    }
+                    gen.close();
+                }
+                gen.close();
+            }
+            gen.close();
+        }
+    }
+
+    private static void appendLoginModules(XmlGenerator gen, String tag, List<LoginModuleConfig> loginModuleConfigs) {
```
[the remark this answers is not in the document]
Comment (tkountis) [reply hazelcast/src/main/java/com/hazelcast/config/ConfigXmlGenerator.java]: Yes, luckily also a test failed with this :+1:

**Pull request merged**